go/tools/builders/constants.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+const (
+	nogoSuccess int = iota
+	nogoError
+	nogoViolation
+)
```

Comment:
Nit: It's a bit weird to see us share a source file like this between 2 binaries.

Should we make a new `tools/nogo` package for the nogo binary and import this as a shared library?

Reply:
Please also add a comment here to note that these are not just internal states, but action exit code (0, 1, 2).

Having the word `exit code` somewhere close would help newer folks grep through the source a bit easier.

Reply:
I added a comment to explain that these are exit codes and another one to explain why (and that) we share the source file. There are ways to break the cyclic dependency, but source file sharing seems simple enough and other targets in this package already to it.